File: android/sdk/src/main/java/app/organicmaps/sdk/util/GzipUtils.java
package app.organicmaps.sdk.util;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.util.zip.GZIPInputStream;
import java.util.zip.GZIPOutputStream;

public final class GzipUtils {
    private GzipUtils() {}

    public static byte[] compress(byte[] data) throws IOException {
        if (data == null || data.length == 0) return data;
        ByteArrayOutputStream bos = new ByteArrayOutputStream(data.length);
        GZIPOutputStream gzip = new GZIPOutputStream(bos);
        gzip.write(data);
        gzip.close();
        byte[] compressed = bos.toByteArray();
        bos.close();
        return compressed;
    }

    public static byte[] decompress(byte[] data) throws IOException {
        if (data == null || data.length == 0) return data;
        ByteArrayInputStream bis = new ByteArrayInputStream(data);
        GZIPInputStream gzip = new GZIPInputStream(bis);
        ByteArrayOutputStream bos = new ByteArrayOutputStream();
        byte[] buffer = new byte[1024];
        int len;
        while ((len = gzip.read(buffer)) > 0) {
            bos.write(buffer, 0, len);
        }
        gzip.close();
        bis.close();
        byte[] decompressed = bos.toByteArray();
        bos.close();
        return decompressed;
    }
}

File: android/app/omaps/src/main/java/app/organicmaps/wear/GzipUtils.kt
package app.organicmaps.wear

import java.io.ByteArrayOutputStream
import java.util.zip.GZIPInputStream

object GzipUtils {
    fun decompress(data: ByteArray): ByteArray {
        val bis = data.inputStream()
        val gis = GZIPInputStream(bis)
        val bos = ByteArrayOutputStream()
        val buffer = ByteArray(1024)
        var len: Int
        while (gis.read(buffer).also { len = it } > 0) {
            bos.write(buffer, 0, len)
        }
        gis.close()
        return bos.toByteArray()
    }
}
